`UV 2.0` installation fails on `aarch64-unknown-linux-gnu` (inside docker basically)

Hello hello! Congrats on the new release! 🎉  

I happen to be building a docker image 10 minutes after the release 😅  and I can see that the installation is failing. 

# Summary
My dockerfile:
```
FROM python:3.11-slim as builder

WORKDIR /app

ENV PYTHONUNBUFFERED 1
ENV PYTHONDONTWRITEBYTECODE 1
ENV PIP_NO_CACHE_DIR 1
ENV RUNTIME docker

RUN apt-get update
RUN apt-get install -y git curl

# install uv
ADD --chmod=755 https://astral.sh/uv/install.sh /install.sh
RUN /install.sh && rm /install.sh
```

then building this, I see this:
```
[+] Building 1.6s (10/13)                                                                                                                                                                                                     docker:orbstack
 => [internal] load build definition from Dockerfile                                                                                                                                                                                     0.0s
 => => transferring dockerfile: 690B                                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/python:3.11-slim                                                                                                                                                                      0.5s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                                                                                                            0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                        0.0s
 => => transferring context: 2B                                                                                                                                                                                                          0.0s
 => [1/8] FROM docker.io/library/python:3.11-slim@sha256:fc39d2e68b554c3f0a5cb8a776280c0b3d73b4c04b83dbade835e2a171ca27ef                                                                                                                0.0s
 => [internal] load build context                                                                                                                                                                                                        0.0s
 => => transferring context: 297B                                                                                                                                                                                                        0.0s
 => CACHED [2/8] WORKDIR /app                                                                                                                                                                                                            0.0s
 => CACHED [3/8] RUN apt-get update                                                                                                                                                                                                      0.0s
 => CACHED [4/8] RUN apt-get install -y git curl                                                                                                                                                                                         0.0s
 => ERROR [5/8] RUN curl -LsSf https://astral.sh/uv/install.sh | sh                                                                                                                                                                      1.0s
------
 > [5/8] RUN curl -LsSf https://astral.sh/uv/install.sh | sh:
0.763 downloading uv 0.2.0 aarch64-unknown-linux-gnu
0.965 curl: (22) The requested URL returned error: 404
0.970 failed to download https://github.com/astral-sh/uv/releases/download/0.2.0/uv-aarch64-unknown-linux-gnu.tar.gz
0.970 this may be a standard network error, but it may also indicate
0.970 that uv's release process is not working. When in doubt
0.970 please feel free to open an issue!
------
Dockerfile:19
--------------------
  17 |     #ADD --chmod=755 https://astral.sh/uv/install.sh /install.sh
  18 |     #RUN /install.sh && rm /install.sh
  19 | >>> RUN curl -LsSf https://astral.sh/uv/install.sh | sh
  20 |
  21 |     COPY requirements.txt .
--------------------
ERROR: failed to solve: process "/bin/sh -c curl -LsSf https://astral.sh/uv/install.sh | sh" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/orbstack/orbstack/w6usv6gxw2sbgk1dzpjkoqahf
make[1]: *** [build] Error 1
make: *** [build-service] Error 2
```

### Steps to Reproduce
1. create the dockerfile I created above: `touch Dockerfile` 
2. build the dockerfile: `docker build -t myimage .` 

## My System
* My mac:
<img width="392" alt="image" src="https://github.com/astral-sh/uv/assets/10492324/e32404c3-8d5a-40a0-a791-b514227a0561">

* My docker version:
```
docker --version
Docker version 25.0.2, build 29cf629
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -215,7 +215,6 @@ unix-archive = ".tar.gz"
 # Target platforms to build apps for (Rust target-triple syntax)
 targets = [
   "aarch64-apple-darwin",
-  "aarch64-unknown-linux-gnu",
   "aarch64-unknown-linux-musl",
   "arm-unknown-linux-musleabihf",
   "armv7-unknown-linux-gnueabihf",

